Master adxl372 add i2c support

stefpopa opened the pull request:
The adxl372 is designed to communicate in either SPI or I2C protocol. It
autodetects the format being used, requiring no configuration control to
select the format.

commodo commented on drivers/iio/accel/Makefile:
```diff
@@ -8,6 +8,7 @@ obj-$(CONFIG_ADXL345) += adxl345_core.o
 obj-$(CONFIG_ADXL345_I2C) += adxl345_i2c.o
 obj-$(CONFIG_ADXL345_SPI) += adxl345_spi.o
 obj-$(CONFIG_ADXL372) += adxl372.o
```
hmm; typically this would be now called `adxl372_core.o` which would require a rename as well

lclausen-adi replied:
I prefer it without the _core.

commodo commented on drivers/iio/accel/adxl372_spi.c:
```diff
@@ -0,0 +1,45 @@
+#include <linux/module.h>
```
i think some header info is required here;
```
 // SPDX-License-Identifier: GPL-2.0+
 /*
  * ADXL372 3-Axis Digital Accelerometer SPI driver
  *
  * Copyright 2018 Analog Devices Inc.
 */
```
and all that

commodo replied:
and in a few other places

stefpopa commented:
Thank you for the review!

Changes in v2:
 * added missing licence headers
 * the header file contains only things shared among modules
 * added a comment which explains the revision ID check
 * returned ENODEV instead of EOPNOTSUPP

commodo commented:
Will merge tomorrow if no other objections
